### PR TITLE
Mix in observer apis to the connector

### PR DESCRIPTION
**Pull request by raymondfeng**
This PR does a few things:

1. Enhance the observer mixin so that it allows around behavior
2. Enable connectors with observer apis

Connect to #464

### CODE REVIEW

File: lib/observer.js [hunk 1]
```diff
@@ -88,11 +88,51 @@ ObserverMixin.notifyObserversOf = function(operation, context, callback) {
...
 ObserverMixin._notifyBaseObservers = function(operation, context, callback) {
   if (this.base && this.base.notifyObserversOf)
     this.base.notifyObserversOf(operation, context, callback);
   else
     callback();
-}
+};
+
+/**
+ * Create an around behavior by notifying before/after observers
```
Comment (ritch) [inline lib/observer.js]: The wording of this is confusing... Not really sure what this does by reading it.

Comment (raymondfeng) [inline lib/observer.js]: Fixed the comment

File: lib/observer.js [hunk 2]
```diff
@@ -88,11 +88,59 @@ ObserverMixin.notifyObserversOf = function(operation, context, callback) {
...
+ *
+ * If an error happens, it fails fast and calls the callback with err.
+ *
+ * @param {String} operation The operation name
+ * @param {Context} context The context object
+ * @param {Function} fn The task to be invoked as fn(done) or fn(context, done)
+ * @param {Function} callback The callback function
+ * @returns {*}
+ */
+ObserverMixin.around = function(operation, context, fn, callback) {
```
Comment (ritch) [inline lib/observer.js]: shouldn't this method name conform to the `notifyObserversOf` convention?

Consider this idea (in the context of your usage):

```js
function work(done) {
  self.connector.commit(self.connection, done);
}

self.notifyObserversBeforeAndAfter('commit', context, work, function(err) {
  // Deference the connection to mark the transaction is not active
  // The connection should have been released back the pool
  self.connection = null;
  cb(err);
});
```

File: test/memory.test.js [hunk 3]
```diff
@@ -584,4 +584,48 @@ describe('Memory connector with options', function() {
 
 });
 
+describe('Memory connector with hookd', function() {
```
Comment (ritch) [inline test/memory.test.js]: hookd => typo?

**Comment by ritch** [comment]
I think we should rename `around` otherwise LGTM

**Pull request merged**